include/matrix3x4SIMD.h

```diff
@@ -503,25 +517,21 @@ struct matrix3x4SIMD
 
 	inline __m128d halfRowAsDouble(size_t _n, bool _0) const
 	{
-		double tmp[2]{
-			_0 ? rows[_n].x : rows[_n].z,
-			_0 ? rows[_n].y : rows[_n].w
-		};
-		return _mm_loadu_pd(tmp);
+		return _mm_cvtps_pd(_0 ? rows[_n].xyxx().getAsRegister() : rows[_n].zwxx().getAsRegister());
 	}
-	static inline __m128d doJob_d(const __m128d& _a0, const __m128d& _a1, const matrix3x4SIMD& _mtx, bool _0)
+	static inline __m128d doJob_d(const __m128d& _a0, const __m128d& _a1, const matrix3x4SIMD& _mtx, bool _xyHalf)
 	{
-		__m128d r0 = _mtx.halfRowAsDouble(0u, _0);
-		__m128d r1 = _mtx.halfRowAsDouble(1u, _0);
-		__m128d r2 = _mtx.halfRowAsDouble(2u, _0);
+		__m128d r0 = _mtx.halfRowAsDouble(0u, _xyHalf);
+		__m128d r1 = _mtx.halfRowAsDouble(1u, _xyHalf);
+		__m128d r2 = _mtx.halfRowAsDouble(2u, _xyHalf);
 
 		const __m128d mask01 = _mm_castsi128_pd(_mm_setr_epi32(0, 0, 0xffffffff, 0xffffffff));
 
 		__m128d res;
 		res = _mm_mul_pd(_mm_shuffle_pd(_a0, _a0, 0), r0);
```

Comment:
could you explain the shuffles to me?

Reply:
Similarly to float32 version. `_a0` and `_a1` are two halves of a row. So:  
* `_mm_shuffle_pd(_a0, _a0, 0)` will give us first element of first half broadcasted (i.e. x brodcasted)  
* `_mm_shuffle_pd(_a0, _a0, 3)` will give us second element of first half broadcasted (y broadcasted). `3` comes from the fact that 3==0b11  
  
Same for second half. The trailing `if` is because in first half it'd have no sense since first half of 3rd row is {0, 0} and so we would have to do something like _mm_add_pd(res, _mm_and_pd(_a1, **mask00**));

Reply:
tested it. Works